app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+#  News Articles
+#
+#  Actions
+#
+#    create:           Create article from data in "new" form
+#    destroy::         Destroy article
+#    edit::            Display form for editing article
+#    index::           List articles, filtered by current query
+#    new::             Display form for new article
+#    show::            Show one article
+#    update::          Update article from "edit" form
+#
+#  Public methods      None (ideally)
+#
+class ArticlesController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :show
+  ]
+  before_action :store_location, except: :destroy
+  before_action :pass_query_params, except: :index
+  before_action :ignore_request_unless_permitted, except: [
+    :index,
+    :show
+  ]
+
+  # ---------- Actions to Display data (index, show, etc.) ---------------------
+
+  # List selected articles, filtered by current Query.
+  # articles  GET /articles(.:format)
+  def index
+    filter_index? ? index_filtered : index_full
+  end
+
+  # article  GET /articles/:id(.:format)
+  def show
+    return false unless (@article = find_or_goto_index(Article, params[:id]))
+
+    @canonical_url = article_url(@article.id)
```

Comment:
By coincidence, it's the same string as the old canonical_url. 
That's not necessarily the case with other controllers, and -- as @pellaea oints out -- might get the bots all excited.